webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala

```diff
@@ -130,6 +130,32 @@ case class PermissionDataGetADM(projectIris: Seq[IRI],
     if (!requestingUser.isSystemUser) throw ForbiddenException("Permission data can only by queried by a SystemUser.")
 }
 
+/**
+ * A message that requests all permissions defined inside a project.
+ * A successful response will be a [[PermissionsForProjectGetResponseADM]].
+ *
+ * @param projectIri     the project for which the permissions are queried.
+ * @param requestingUser the user initiation the request.
+ * @param apiRequestID   the API request ID.
+ */
+case class PermissionsForProjectGetRequestADM(projectIri: IRI,
+                                                requestingUser: UserADM,
+                                                apiRequestID: UUID
+                                               ) extends PermissionsResponderRequestADM {
+
+    implicit protected val stringFormatter: StringFormatter = StringFormatter.getInstanceForConstantOntologies
+    stringFormatter.validateProjectIri(projectIri, throw BadRequestException(s"Invalid project IRI"))
+
+    // Check user's permission for the operation
+    if (!requestingUser.isSystemAdmin
+        && !requestingUser.permissions.isProjectAdmin(projectIri)
+        && !requestingUser.isSystemUser
```

Comment:
There is no `systemUser` outside of DSP-API, so he should not be able to come over this route.

Reply:
Can you please elaborate on this?

Reply:
So the only problem here is that the check `!requestingUser.isSystemUser` is unnecessary, right?

Reply:
Yes. If we want to be extra careful, then detecting ˋsystemUserˋ should lead to throwing a forbidden exception.